builtin/credential/aws/path_config_identity.go

```diff
@@ -59,24 +90,41 @@ func pathConfigIdentityUpdate(ctx context.Context, req *logical.Request, data *f
 		if !strutil.StrListContains(allowedIAMAliasValues, iamAlias) {
 			return logical.ErrorResponse(fmt.Sprintf("iam_alias of %q not in set of allowed values: %v", iamAlias, allowedIAMAliasValues)), nil
 		}
-		configEntry.IAMAlias = iamAlias
-		entry, err := logical.StorageEntryJSON("config/identity", configEntry)
-		if err != nil {
-			return nil, err
-		}
-		if err := req.Storage.Put(ctx, entry); err != nil {
-			return nil, err
+		config.IAMAlias = iamAlias
+	}
+
+	ec2AliasRaw, ok := data.GetOk("ec2_alias")
+	if ok {
+		ec2Alias := ec2AliasRaw.(string)
+		allowedEC2AliasValues := []string{identityAliasEC2InstanceID, identityAliasEC2ImageID}
+		if !strutil.StrListContains(allowedEC2AliasValues, ec2Alias) {
+			return logical.ErrorResponse(fmt.Sprintf("ec2_alias of %q not in set of allowed values: %v", ec2Alias, allowedEC2AliasValues)), nil
 		}
+		config.EC2Alias = ec2Alias
+	}
+
+	entry, err := logical.StorageEntryJSON("config/identity", config)
+	if err != nil {
+		return nil, err
+	}
+
+	err = req.Storage.Put(ctx, entry)
+	if err != nil {
+		return nil, err
 	}
+
 	return nil, nil
 }
 
 type identityConfig struct {
 	IAMAlias string `json:"iam_alias"`
+	EC2Alias string `json:"ec2_alias"`
 }
 
 const identityAliasIAMUniqueID = "unique_id"
 const identityAliasIAMFullArn = "full_arn"
+const identityAliasEC2InstanceID = "instance-id"
+const identityAliasEC2ImageID = "image-id"
```

Comment:
Can we make these `_` to match the IAM values?

Reply:
Fixed.